### PR TITLE
[Ubuntu] parse java semver from API

### PR DESCRIPTION
**Pull request by mikhailkoliada**
# Description

As we are parsing semver from the API server on macOS and Windows it is the easiest way to do so on Linux, too

## Check list
- [X] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [X] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/linux/scripts/installers/java-tools.sh [hunk 1]
```diff
@@ -67,24 +67,20 @@ installOpenJDK() {
...
+        assetUrl=$(curl -s "https://api.adoptopenjdk.net/v3/assets/latest/${JAVA_VERSION}/hotspot")
     else
         echo "${VENDOR_NAME} is invalid, valid names are: Temurin-Hotspot and Adopt"
         exit 1
     fi
 
-    JAVA_TOOLCACHE_PATH="${AGENT_TOOLSDIRECTORY}/Java_${VENDOR_NAME}_jdk"
-
-    fullJavaVersion=$(cat "${javaVersionPath}/release" | grep "^SEMANTIC" | cut -d "=" -f 2 | tr -d "\"" | tr "+" "-")
+    asset=$(echo ${assetUrl} | jq -r '.[] | select(.binary.os=="linux" and .binary.image_type=="jdk" and .binary.architecture=="x64")')
```
Comment (al-cheb) [inline images/linux/scripts/installers/java-tools.sh:80]: How can we be sure that version obtained from the link is the same we install via apt ?

Comment (mikhailkoliada) [reply images/linux/scripts/installers/java-tools.sh:80]: @al-cheb I think we can introduce the comparison between `java -version` and the api

➜ echo $asset | jq -r '.version.openjdk_version'
17.0.4.1+1

File: images/linux/scripts/installers/java-tools.sh [hunk 2]
```diff
@@ -67,24 +67,20 @@ installOpenJDK() {
     if [[ ${VENDOR_NAME} == "Temurin-Hotspot" ]]; then
         apt-get -y install temurin-${JAVA_VERSION}-jdk=\*
         javaVersionPath="/usr/lib/jvm/temurin-${JAVA_VERSION}-jdk-amd64"
+        assetUrl=$(curl -s "https://api.adoptium.net/v3/assets/latest/${JAVA_VERSION}/hotspot")
```
Comment (Smithbull001) [inline images/linux/scripts/installers/java-tools.sh:70]: Well done

File: images/linux/scripts/installers/java-tools.sh [hunk 3]
```diff
@@ -67,24 +67,20 @@ installOpenJDK() {
...
     elif [[ ${VENDOR_NAME} == "Adopt" ]]; then
         apt-get -y install adoptopenjdk-${JAVA_VERSION}-hotspot=\*
         javaVersionPath="/usr/lib/jvm/adoptopenjdk-${JAVA_VERSION}-hotspot-amd64"
+        assetUrl=$(curl -s "https://api.adoptopenjdk.net/v3/assets/latest/${JAVA_VERSION}/hotspot")
     else
         echo "${VENDOR_NAME} is invalid, valid names are: Temurin-Hotspot and Adopt"
         exit 1
     fi
```
Comment (Smithbull001) [inline images/linux/scripts/installers/java-tools.sh:79]: Yes

File: images/linux/scripts/installers/java-tools.sh (hunk 1 above)
Comment (Smithbull001) [reply images/linux/scripts/installers/java-tools.sh:80]: Awesome

**Pull request closed without merging**